src/SourceBuild/content/Directory.Build.props

```diff
@@ -206,6 +206,13 @@
     <XPlatSourceBuildTasksAssembly>$([MSBuild]::NormalizePath('$(ArtifactsBinDir)', 'Microsoft.DotNet.SourceBuild.Tasks.XPlat', '$(Configuration)', 'Microsoft.DotNet.SourceBuild.Tasks.XPlat.dll'))</XPlatSourceBuildTasksAssembly>
     <LeakDetectionTasksAssembly>$([MSBuild]::NormalizePath('$(ArtifactsBinDir)', 'Microsoft.DotNet.SourceBuild.Tasks.LeakDetection', '$(Configuration)', 'Microsoft.DotNet.SourceBuild.Tasks.LeakDetection.dll'))</LeakDetectionTasksAssembly>
     <SdkArchiveDiffTasksAssembly>$([MSBuild]::NormalizePath('$(ArtifactsBinDir)', 'Microsoft.DotNet.SourceBuild.Tasks.SdkArchiveDiff', '$(Configuration)', 'Microsoft.DotNet.SourceBuild.Tasks.SdkArchiveDiff.dll'))</SdkArchiveDiffTasksAssembly>
+    <PublishVerticalManifestAssembly>$([MSBuild]::NormalizePath('$(ArtifactsBinDir)', 'Microsoft.DotNet.SourceBuild.Tasks.PublishVerticalManifest', '$(Configuration)', 'Microsoft.DotNet.SourceBuild.Tasks.PublishVerticalManifest.dll'))</PublishVerticalManifestAssembly>
+  </PropertyGroup>
+
+  <!-- The predefined environmental variable `BUILD_BUILDNUMBER` is getting overwritten in repo-projects,
+    so save it in a different varialbe-->
+  <PropertyGroup>
+    <VmrBuildNumber>$(BUILD_BUILDNUMBER)"</VmrBuildNumber>
   </PropertyGroup>
```

Comment:
I would move this into build.proj into the new target.

Reply:
Will that work? We're overwriting `BUILD_BUILDNUMBER` in repo-projects `Directory.Targets`. I originally tried not introducing a new property at all, just using `BUILD_BUILDNUMBER` but that didn't work

Reply:
Yes. The env var override only applies to projects inside the repo-projects folder. Build.proj doesn't import that infrastructure as it lives outside that folder.

Reply:
That's how I understood it too, but when I tried what I mentioned above, `BUILD_BUILDNUMBER` was reset to an empty property. I'll try what you recommended

Reply:
Looks like it didn't work. When it exists the repo-projects, the `BUILD_BUILDNUMBER` gets lost. I'll save it in a property in the base `Directory.Build.Props`